### PR TITLE
Ma/ds overview of all locks

### PR DESCRIPTION
**Pull request by medo-freiheit**
[Story link](https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/-M37cZx7XYoOvQ5Mmcir/STORY/-MqJ6Q5ajTW2cc3CDRw_/)

### CODE REVIEW

File: services/frontend-service/src/ui/AppBar/LocksDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+import WarningRoundedIcon from '@material-ui/icons/WarningRounded';
+import ExpandMoreRounded from '@material-ui/icons/ExpandMoreRounded';
+import { Button, Box, Drawer, Grid, Paper, Typography, Divider, List } from '@material-ui/core';
+
+import { GetOverviewResponse, Commit } from '../../api/api';
+import { theme } from '../App/styles';
+
+import { useStyles } from './styles';
+
+interface ILocks {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:28]: remove the `I` prefix please, we don't use that anywhere else.

File: services/frontend-service/src/ui/AppBar/LocksDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        );
+    }
+
+    return (
+        <List className={classes.environments} sx={{ width: 'auto' }}>
+            {locks.map((lock) => (
+                <>
+                    <Grid item xs={12} key={lock.id} onClick={onClick}>
+                        <Paper className="environment">
+                            {(!!lock.age && lock.age) > 2 && <WarningRoundedIcon color="error" />}
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:54]: This seems like a lot of logic in one line, please extract a variable, e.g.
````
const isLockOld = lock.age... > 2
const warning = isLockOld ? <Warning.../> : null
```
`{warning}`

File: services/frontend-service/src/ui/AppBar/LocksDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        }
+        setState({ isOpen: open });
+    };
+
+    const calcAge = useCallback((time: any) => {
+        if (!time) return -1;
+        const curTime = new Date().getTime();
+        const diffTime = curTime - time;
+        const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+        return diffDays;
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:109]:
```suggestion
        const msPerDay = 1000 * 60 * 60 * 24;
        const diffDays = Math.ceil(diffTime / (msPerDay));
        return diffDays;
```

File: services/frontend-service/src/ui/AppBar/LocksDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                        id: value,
+                        ...app.locks[value],
+                        type: app.name,
+                        age: calcAge(app.locks[value].commit?.authorTime),
+                    })),
+                ];
+            }
+        }
+        nwLocks.sort((a: ILocks, b: ILocks) => {
+            if (!a.commit?.authorTime || !b.commit?.authorTime) return 0;
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:138]: why is the `!` necessary? We don't have null locks, or do we?

File: services/frontend-service/src/ui/AppBar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+Copyright 2021 freiheit.com*/
+import * as React from 'react';
+import Box from '@material-ui/core/Box';
+import AppBar from '@material-ui/core/AppBar';
+import Typography from '@material-ui/core/Typography';
+import { AppDrawer } from './AppDrawer';
+import * as api from '../../api/api';
+import { LocksDrawer } from './LocksDrawer';
+
+export const Header: React.FC<any> = (props: { overview: api.GetOverviewResponse }) => {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/AppBar/index.tsx:25]: if the only thing in this file is this component, then the file should be named `Header.tsx`

File: services/frontend-service/src/ui/AppBar/LocksDrawer.tsx (hunk 4 above)
Comment (medo-freiheit) [reply services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:138]: Not the lock but the commit and the authorTime can be null

File: services/cd-service/pkg/service/overview.go [hunk 6]
```diff
@@ -70,8 +70,15 @@ func (o *OverviewServiceServer) getOverview(
 				return nil, err
 			} else {
 				for lockId, lock := range locks {
+					lockCommit := (*api.Commit)(nil)
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/service/overview.go:73]: this should be something like
`var lockCommit *api.Commit = nil`

File: services/cd-service/pkg/service/overview.go [hunk 7]
```diff
@@ -121,8 +128,15 @@ func (o *OverviewServiceServer) getOverview(
 						return nil, err
 					} else {
 						for lockId, lock := range appLocks {
+							lockCommit := (*api.Commit)(nil)
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/service/overview.go:131]: same here

File: services/frontend-service/src/ui/AppBar/LocksDrawer.tsx (hunk 4 above)
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:149]: I think in that case you can remove the `!`. (unless the compiler complains)

Comment (medo-freiheit) [reply services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:149]: Yes, the compiler is complaining about it

Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/AppBar/LocksDrawer.tsx:149]: huh, weird, but ok, then keep it.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**